### PR TITLE
#573 - update http

### PR DESCRIPTION
**Pull request by olenagerasimova**
Updated http to `0.16.2`, gem and docker adapters and corrected `YamlPermissions` to check permissions for user groups.

### CODE REVIEW

File: src/main/java/com/artipie/LoggingPermissions.java [hunk 1]
```diff
@@ -63,12 +64,12 @@ public LoggingPermissions(final Permissions origin, final Level level) {
...
-    public boolean allowed(final String name, final String action) {
-        final boolean res = this.origin.allowed(name, action);
+    public boolean allowed(final Authentication.User user, final String action) {
+        final boolean res = this.origin.allowed(user, action);
         if (res) {
-            Logger.log(this.level, this.origin, "Operation '%s' allowed for '%s'", action, name);
+            Logger.log(this.level, this.origin, "Operation '%s' allowed for '%s'", action, user);
         } else {
-            Logger.log(this.level, this.origin, "Operation '%s' denied for '%s'", action, name);
+            Logger.log(this.level, this.origin, "Operation '%s' denied for '%s'", action, user);
```
Comment (olegmoz) [inline src/main/java/com/artipie/LoggingPermissions.java:72]: @olenagerasimova I think `User` class lacks proper `toString` method at the moment. So we need to implement `User` formatting ourselves here or just print user's name for now

File: src/test/java/com/artipie/YamlPermissionsTest.java [hunk 2]
```diff
@@ -48,10 +53,11 @@ void johnCanDownloadDeployAndDelete() throws Exception {
...
                 new ListOf<org.hamcrest.Matcher<? super YamlPermissions>>(
-                    new MatcherOf<>(perm -> { return perm.allowed(uname, "delete"); }),
-                    new MatcherOf<>(perm -> { return perm.allowed(uname, "deploy"); }),
-                    new MatcherOf<>(perm -> { return perm.allowed(uname, "download"); }),
-                    new MatcherOf<>(perm -> !perm.allowed(uname, "install"))
+                    // @checkstyle LineLengthCheck (4 lines)
+                    new MatcherOf<>(perm -> { return perm.allowed(new Authentication.User(uname), "delete"); }),
+                    new MatcherOf<>(perm -> { return perm.allowed(new Authentication.User(uname), "deploy"); }),
+                    new MatcherOf<>(perm -> { return perm.allowed(new Authentication.User(uname), "download"); }),
+                    new MatcherOf<>(perm -> !perm.allowed(new Authentication.User(uname), "install"))
```
Comment (olegmoz) [inline src/test/java/com/artipie/YamlPermissionsTest.java:60]: @olenagerasimova I'd suggest to extract variable for `Authentication.User(uname)` to make code easier to read here and in `janeCanDownloadAndDeploy`, `adminCanDoAnything` methods

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see two comments

File: src/main/java/com/artipie/LoggingPermissions.java (hunk 1 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/LoggingPermissions.java:72]: @olegmoz created https://github.com/artipie/http/issues/246

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#641` is not assigned, can't get performer

**Pull request merged**